src/Plugin.php

```diff
@@ -86,6 +93,45 @@ public static function upload_mime_types(array $mimes) {
   }
 
   /**
+   * Removes SVG image size attributes from the content.
+   *
+   * @implements the_content
+   */
+  public static function the_content($content) {
+    $content = preg_replace_callback('@<img [^>]*src=[^>]+?.svg[^>]*>@', function ($matches) {
+      return static::removeSizeAttributes($matches[0]);
+    }, $content);
+    return $content;
+  }
+
+  /**
+   * Removes size attributes from SVG images.
+   *
+   * @implements wp_get_attachment_metadata
+   * @implements image_send_to_editor
+   * @implements post_thumbnail_html
+   */
+  public static function removeSvgSizeAttributes($data, $post_id, $attachment_id = NULL) {
+    if ($data && get_post_mime_type($attachment_id ?? $post_id) === 'image/svg+xml') {
+      $data = static::removeSizeAttributes($data);
+    }
+    return $data;
+  }
+
+  /**
+   * Removes size attributes from a given string or array.
+   */
+  public static function removeSizeAttributes($data) {
+    if (is_array($data)) {
+      unset($data['width'], $data['height']);
```

Comment:
What you're doing here is called "function argument overloading" and generally discouraged, because the lack of type safety makes the code much harder to read and understand.  Two separate functions, of which one takes an array, and the other takes a string would be better.  (Given that the array only happens in a single case, you do not even need a helper function.)

If you want to keep the current code, we should at least document that the array only happens for the hook `wp_get_attachment_metadata`.